File: src/com/jupiter/ganymede/math/regression/Regressor.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.jupiter.ganymede.math.regression;

import java.util.Set;
import java.util.function.Function;

/**
 *
 * @author Nathan Templon
 * @param <T>
 * @param <F>
 */
@FunctionalInterface
public interface Regressor<T extends Number, F extends Number> {
    
    Function<T, F> bestFit(Set<Point<? extends T, ? extends F>> points);
    
    
    // Nested Classes
    public final static class Point<G extends Number, H extends Number> {
        
        // Fields
        public final G first;
        public final H second;
        
        
        // Intialization
        public Point(G first, H second) {
            this.first = first;
            this.second = second;
        }
        
    }
    
}

File: src/com/jupiter/ganymede/math/regression/Regressor.kt
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.jupiter.ganymede.math.regression

import java.util.function.Function

/**

 * @author Nathan Templon
 * *
 * @param
 * *
 * @param
 */
@FunctionalInterface
public interface Regressor<T : Number, F : Number> {

    public fun bestFit(points: Set<Point<T, F>>): Function<T, F>


    // Nested Classes
    data public class Point<G : Number, H : Number>(public val first: G, public val second: H)

}
